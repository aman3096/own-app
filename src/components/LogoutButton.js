import React from 'react';
import {Button} from './Styles';
import {useAuth0} from '@auth0/auth0-react';

const LogoutButton = () => {
  const {isAuthenticated, logout} = useAuth0()
 return(

  isAuthenticated && (
   <Button data-testid="logoutbtn" onClick={() => logout()}>
      Log Out
   </Button>
   )
 )
}

export default LogoutButton