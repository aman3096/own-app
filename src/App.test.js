import { render, screen,cleanup } from '@testing-library/react';
import App from './App';
import Navbar from './components/Navbar';
import Profile from './components/Profile'
import LogoutButton from './components/LogoutButton';
import LoginButton from './components/LoginButton';

afterEach(cleanup)

it('should take a snapshot of App', () => {
  const { asFragment } = render(<App />)
  
  expect(asFragment(<App />)).toMatchSnapshot()
 })

 it('should take a snapshot of Navbar', () => {
  const { asFragment } = render(<Navbar />)
  
  expect(asFragment(<Navbar />)).toMatchSnapshot()
 })

 it('should take a snapshot of Profile', () => {
  const { asFragment } = render(<Profile />)
  
  expect(asFragment(<Profile />)).toMatchSnapshot()
 })

test('must contain Login button at start',() => {
  render(<App/>);
  const loginBtn = screen.getByText(/Log In/i);
  expect(loginBtn).toBeInTheDocument();

})

describe('App', () => {
  test('renders App component', () => {
    render(<App />);
    
    screen.debug();
  });
});

describe('Navbar', () => {
  test('renders Navbar component', () => {
    render(<Navbar/>);

    screen.getByText('Log in')
    screen.debug();
  });
});

describe('Profile', () => {
  test('renders Profile component', () => {
    render(<Profile/>);
    screen.debug();
  });
});
describe('Log in', () => {
  test('renders Log in component', () => {
    render(<LoginButton/>);
    screen.debug()

    
  });
});




