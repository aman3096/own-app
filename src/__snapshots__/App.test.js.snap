// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of  1`] = `
<DocumentFragment>
  <nav
    class="sc-gsTCUz kaNQOD"
  >
    <button
      class="sc-bdfBwQ kqdBOk"
      data-testid="mainButton"
    >
      Log in
    </button>
  </nav>
</DocumentFragment>
`;

exports[`should take a snapshot of App 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="sc-gsTCUz kaNQOD"
    >
      <button
        class="sc-bdfBwQ kqdBOk"
        data-testid="mainButton"
      >
        Log in
      </button>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`should take a snapshot of Navbar 1`] = `
<DocumentFragment>
  <nav
    class="sc-gsTCUz kaNQOD"
  >
    <button
      class="sc-bdfBwQ kqdBOk"
      data-testid="mainButton"
    >
      Log in
    </button>
  </nav>
</DocumentFragment>
`;

exports[`should take a snapshot of Navbar 2`] = `
<DocumentFragment>
  <nav
    class="sc-gsTCUz kaNQOD"
  >
    <button
      class="sc-bdfBwQ kqdBOk"
      data-testid="mainButton"
    >
      Log in
    </button>
  </nav>
</DocumentFragment>
`;

exports[`should take a snapshot of Profile 1`] = `<DocumentFragment />`;
